duckduckgo/duckduckgo-privacy-extension
# Tracker blocking opt in experiment

**Reviewer:** @jdorweiler 

**CC:** @russellholt @bbraithwaite @bsstoner @chrismorast @MariagraziaAlastra 
**Depends on:** https://dub.duckduckgo.com/duckduckgo/ddg/pull/15986

## Description:
This is part of a larger effort to measure the impact of making tracker blocking opt-in in the extension and apps. This PR contains a minimal experiment framework based on atb variant, a content script that is injected into the post install page, changes to the popup to show an 'activate tracker blocking' screen when it's disabled, and a lot of functionality around when tracker blocking is turned on or off.

## Steps to test this PR:
1. Follow testing steps 1-10 in https://dub.duckduckgo.com/duckduckgo/ddg/pull/15986 to get the extension installed in the experiment group.
2. There should be a red badge on the extension icon when tracker blocking is disabled.
3. There should also be an 'activate tracker blocking' CTA in the extension popup when tracker blocking is disabled.
4. Trackers should be tallied in popup but not blocked when tracker blocking is disabled.
5. There are four places where tracker blocking can be activated: in the extension popup, on the post install page (button in middle of page, button in header that appears when you scroll down on post install page, toggle on extension settings page).
6. When tracker blocking is turned on via any of these methods, all extension icons should update to hide the red badge and all popups should update to hide the activate tracker blocking CTA. When tracker blocking is enabled while on the post install page via either in page buttons or the popup, a `tba` cookie should be set. This is to remember that tracker blocking has been enabled so the post install page shows the success state. We do this so that if a user installs then opens the popup on the post install page and activates tracker blocking, the post install page reloads and shows the success state. If you want to test other CTAs on the post install page, make sure to delete this cookie first.
7. When tracker blocking is enabled via the popup CTA, the current page should reload after a few seconds.
8. When tracker blocking is disabled, 'report broken site' button should not appear in the hamburger menu in the popup.
9. All of this should only affect new installs in the experiment group (va=k). Make sure nothing here affects existing users. Test with control group (va=l) and other or no variant / atb variant to make sure none of these changes impact those groups.
10. Check pixels if you have time, for reference see https://app.asana.com/0/0/1122176086265607/f

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
